util/update.go

```diff
@@ -0,0 +1,92 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-version"
+)
+
+type githubResponse struct {
+	Name string `json:"name"`
+}
+
+// CheckForRigUpdate checks to see if an upgrdate to rig is available, if so, return a message
+func CheckForRigUpdate(curRigVersion string) string {
+	// Local dev, version == "master" which isn't going to parse.
+	curVer, verr := version.NewVersion(curRigVersion)
+	if tag, err := currentRigReleaseTag(); err == nil {
+		if tagVer, verr2 := version.NewVersion(tag); verr2 == nil {
+			if verr != nil || tagVer.Compare(curVer) > 0 {
+				return "An update for rig is available: " + tag
+			}
+		}
+	} else {
+		if Logger().IsVerbose {
```

Comment:
If you just use `Logger().Verbose(...)` you don't have to do a verbose check.  In fact, it's incorrect to hide a warning behind verbose, verbose it literally for extra noisy messages, a warning does not categorize as chatty.

Reply:
Thanks. I was on the fence about reporting errors/warning in general and took the approach "if you want to see them, put in -v." I expected most people wouldn't care that the update check didn't work (so I was just failing silently). Trying to reduce distractions/support issues for what would be a bug in the program.